sdk/storage/azure-storage-blob/azure/storage/blob/blob_client.py

```diff
@@ -133,6 +133,7 @@ def __init__(
         except AttributeError:
             raise ValueError("Blob URL must be a string.")
         parsed_url = urlparse(blob_url.rstrip('/'))
+        self.account_name = parsed_url.hostname.split('.')[0]
```

Comment:
>parsed_url.hostname.split('.')[0] [](start = 28, length = 33)

Not going to work with custom domains

Reply:
is there any pattern we can follow for customized domain?

Reply:
I don't think so - a customized domain could be supplied with no key from which to generate a sas. An error is already raised in `get_blob_shared_access_signature` in this scenario, so you shouldn't need to do additional work for this.

Reply:
while I think account_name is required to generate sas token, if users are using oauth token as credential, then account_name cannot be parsed out right?